tests/console/sshd.pm

```diff
@@ -97,45 +92,55 @@ sub run {
-    assert_script_run "install -m 0644 -o $ssh_testman ~/.ssh/id_rsa.pub /home/$ssh_testman/.ssh/authorized_keys";
-    assert_script_run "cat /home/$ssh_testman/.ssh/id_rsa.pub >> /home/$ssh_testman/.ssh/authorized_keys";
 
-    # Test non-interactive SSH and after that remove RSA keys
+    # Make sure user has both public keys in authorized_keys
+    assert_script_run "su -c \"cp /home/$ssh_testman/.ssh/{id_rsa.pub,authorized_keys}\"";
+    assert_script_run "cat ~/.ssh/id_rsa.pub >> /home/$ssh_testman/.ssh/authorized_keys";
+
+    # Test non-interactive SSH
     assert_script_run "ssh -4v $ssh_testman\@localhost bash -c 'whoami | grep $ssh_testman'";
 
     # Port forwarding (bsc#1131709 bsc#1133386)
-    assert_script_run "( ssh -NL 4242:localhost:22 $ssh_testman\@localhost & )";
-    assert_script_run "( ssh -NR 0.0.0.0:5252:localhost:22 $ssh_testman\@localhost & )";
+    assert_script_run "echo 'sshd.pm: Testing port forwarding' | logger";
+    assert_script_run "( ssh -vNL 4242:localhost:22 $ssh_testman\@localhost & )";
+    assert_script_run "( ssh -vNR 0.0.0.0:5252:localhost:22 $ssh_testman\@localhost & )";
     assert_script_run 'until ss -tulpn|grep sshd|egrep "4242|5252";do sleep 1;done';
+
+    # Scan public keys on forwarded ports
     assert_script_run "ssh-keyscan -p 4242 localhost >> ~/.ssh/known_hosts";
     assert_script_run "ssh-keyscan -p 5252 localhost >> ~/.ssh/known_hosts";
 
-    assert_script_run "ssh -p 4242 $ssh_testman\@localhost whoami";
-    assert_script_run "ssh -p 5252 $ssh_testman\@localhost whoami";
-    assert_script_run "ssh -p 4242 $ssh_testman\@localhost 'ssh-keyscan -p 22 localhost 127.0.0.1 ::1 >> ~/.ssh/known_hosts'";
-    assert_script_run "ssh -p 4242 $ssh_testman\@localhost 'ssh-keyscan -p 4242 localhost 127.0.0.1 ::1 >> ~/.ssh/known_hosts'";
-    assert_script_run "ssh -p 4242 $ssh_testman\@localhost 'ssh-keyscan -p 5252 localhost 127.0.0.1 ::1 >> ~/.ssh/known_hosts'";
+    # Connect to forwarded ports
+    assert_script_run "ssh -v -p 4242 $ssh_testman\@localhost whoami";
+    assert_script_run "ssh -v -p 5252 $ssh_testman\@localhost whoami";
+
+    # Copy the list of known hosts to $ssh_testman's .ssh directory
+    assert_script_run "install -m 0400 -o $ssh_testman ~/.ssh/known_hosts /home/$ssh_testman/.ssh/known_hosts";
+
+    # Test SSH command within SSH command
+    assert_script_run "ssh -v -p 4242 -tt $ssh_testman\@localhost ssh -tt $ssh_testman\@localhost whoami";
 
-    assert_script_run "ssh -p 4242 -tt $ssh_testman\@localhost ssh -tt $ssh_testman\@localhost whoami";
-    assert_script_run "ssh -t -o ProxyCommand='ssh $ssh_testman\@localhost nc localhost 4242' $ssh_testman\@localhost whoami";
+    # Test ProxyCommand option
+    assert_script_run "ssh -v -t -o ProxyCommand='ssh -v $ssh_testman\@localhost nc localhost 4242' $ssh_testman\@localhost whoami";
```

Comment:
Just for fun, I have found a crazy bashy way of how to replace `netcat` call.  :D

```
(exec 3<>/dev/tcp/localhost/4242; cat >&3; cat <&3; exec 3<&-)
```

Maybe for the future readiness improvement you can use [quote like operators](https://perldoc.perl.org/perlop#Quote-Like-Operators) to avoid escaping.

Reply:
Quote like operators for sure.

I will do some research about the `netcat` replacement.

Let's keep this thread as a reminder then. Thenk you!